File: Carthage/Checkouts/SwiftTheme/OCDemo/SelectThemeViewController.m
//
//  SelectThemeViewController.m
//  SwiftTheme
//
//  Created by Gesen on 16/5/27.
//  Copyright © 2016年 Gesen. All rights reserved.
//

#import "SelectThemeViewController.h"
#import "Global.h"
#import "MyThemes.h"
#import "OCDemo-Swift.h"

@interface SelectThemeViewController ()

@end

@implementation SelectThemeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.theme_backgroundColor = globalBackgroundColorPicker;
}

- (IBAction)tapRed:(id)sender {
    [MyThemes switchTo:MyThemesTypeRed];
}

- (IBAction)tapYellow:(id)sender {
    [MyThemes switchTo:MyThemesTypeYellow];
}

- (IBAction)tapBlue:(id)sender {
    [MyThemes switchTo:MyThemesTypeBlue];
}

@end

File: Carthage/Checkouts/SwiftTheme/PlistDemo/SelectThemeViewController.swift
//
//  SelectThemeViewController.swift
//  PlistDemo
//
//  Created by Gesen on 16/3/1.
//  Copyright © 2016年 Gesen. All rights reserved.
//

import UIKit

class SelectThemeViewController: UIViewController, UIAlertViewDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()

        view.theme_backgroundColor = "Global.backgroundColor"
    }
    
    @IBAction func tapRed(_ sender: AnyObject) {
        MyThemes.switchTo(.red)
    }
    
    @IBAction func tapYellow(_ sender: AnyObject) {
        MyThemes.switchTo(.yello)
    }
    
    @IBAction func tapBlue(_ sender: AnyObject) {
        guard MyThemes.isBlueThemeExist() else {

            let title   = "Not Downloaded"
            let message = "Download the theme right now?"
            
            if #available(iOS 8.0, *) {
                
                let alert = UIAlertController(title: title,
                                              message: message,
                                              preferredStyle: .alert)

                alert.addAction(UIAlertAction(
                    title: "Cancel",
                    style: .cancel,
                    handler: nil)
                )
                alert.addAction(UIAlertAction(
                    title: "Sure",
                    style: .default,
                    handler: { [unowned self] _ in
                        self.downloadStart()
                    })
                )
                
                present(alert, animated: true, completion: nil)
                
            } else {
                
                UIAlertView(title: title,
                            message: message,
                            delegate: self,
                            cancelButtonTitle: "Cancel",
                            otherButtonTitles: "Sure").show()
                
            }
            return
        }
        
        MyThemes.switchTo(.blue)
    }
    
    func alertView(_ alertView: UIAlertView, clickedButtonAt buttonIndex: Int) {
        if buttonIndex == 1 {
            downloadStart()
        }
    }
    
    fileprivate func downloadStart() {
        let HUD = MBProgressHUD.showHUD("Download Theme...")
        
        MyThemes.downloadBlueTask() { isSuccess in
            HUD.labelText = isSuccess ? "Successful!" : "Failure!"
            HUD.mode = .text
            HUD.hide(true, afterDelay: 1)
            
            if isSuccess {
                MyThemes.switchTo(.blue)
            }
        }
    }

}
